encord/orm/dataset.py

```diff
@@ -54,41 +57,189 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+class ImageData:
```

Comment:
[curious]🤔 

Any reason not to use a `dataclass` here? seem like it would simplify the implementation and will be consistent with most of the file.

Reply:
Yeah actually there is. 
* Similar to DataRow not being a dataclass, it makes it easier to define e.g. read only properties, if properties are not updatable. (the user should not feel like they can update the `file_size` as it makes no sense)
* It could very well be that we extend this like the DataRow and allow it to fetch the signed url itself. In this case, it would not really be a dataclass anymore.

Are these reasons convincing?

Reply:
Yeah totally makes sense 👍